### PR TITLE
worktree:  add functions to get name and path

### PR DESCRIPTION
**Pull request by mkeeler**
This is a reworking of my [previous PR](https://github.com/libgit2/libgit2/pull/4639) to clean up some commit history

This adds two new functions:

* `git_worktree_name` - retrieves the name from a `git_worktree` 
* `git_worktree_path` - retrieves the location on the file system from a `git_worktree`

### CODE REVIEW

File: include/git2/worktree.h [hunk 1]
```diff
@@ -148,6 +148,24 @@ GIT_EXTERN(int) git_worktree_unlock(git_worktree *wt);
  */
 GIT_EXTERN(int) git_worktree_is_locked(git_buf *reason, const git_worktree *wt);
 
+/**
+ * Retrieve the name of the worktree
+ *
+ * @param wt Worktree to get the name for
+ * @return The worktree's name. The pointer returned is valid for the 
+ *  lifetime of the git_worktree
```
Comment (tiennou) [inline include/git2/worktree.h]: Nitpick: still a stray space here.

File: include/git2/worktree.h [hunk 2]
```diff
@@ -148,6 +148,24 @@ GIT_EXTERN(int) git_worktree_unlock(git_worktree *wt);
...
+ */
+GIT_EXTERN(const char *) git_worktree_name(const git_worktree *wt);
+
+/**
+ * Retrieve the filesystem path for the worktree
+ *
+ * @param wt Worktree to get the path for
+ * @return The worktree's filesystem path. The pointer returned 
+ *  is valid for the lifetime of the git_worktree.
```
Comment (tiennou) [inline include/git2/worktree.h]: Ditto.

File: src/worktree.c [hunk 3]
```diff
@@ -455,6 +457,18 @@ int git_worktree_is_locked(git_buf *reason, const git_worktree *wt)
 	return ret;
 }
 
+const char * git_worktree_name(const git_worktree *wt)
```
Comment (tiennou) [inline src/worktree.c]: Nitpick: no space after `*`.

File: src/worktree.c [hunk 4]
```diff
@@ -455,6 +457,18 @@ int git_worktree_is_locked(git_buf *reason, const git_worktree *wt)
...
+{
+	assert(wt);
+	return wt->name;
+}
+
+const char * git_worktree_path(const git_worktree *wt)
```
Comment (tiennou) [inline src/worktree.c]: Ditto.

File: tests/clar/sandbox.h [hunk 5]
```diff
@@ -31,13 +35,21 @@ find_tmp_path(char *buffer, size_t length)
 			continue;
 
 		if (is_valid_tmp_path(env)) {
+#ifdef __APPLE__
+			if ( length >= PATH_MAX && realpath(env, buffer) != NULL )
```
Comment (tiennou) [inline tests/clar/sandbox.h]: Nitpick: no spaces inside braces.

File: tests/clar/sandbox.h [hunk 6]
```diff
@@ -31,13 +35,21 @@ find_tmp_path(char *buffer, size_t length)
...
+#endif
 			strncpy(buffer, env, length);
 			return 0;
 		}
 	}
 
 	/* If the environment doesn't say anything, try to use /tmp */
 	if (is_valid_tmp_path("/tmp")) {
+#ifdef __APPLE__
+		if ( length >= PATH_MAX && realpath("/tmp", buffer) != NULL )
```
Comment (tiennou) [inline tests/clar/sandbox.h]: Ditto.

File: tests/worktree/worktree.c [hunk 7]
```diff
@@ -355,6 +355,26 @@ void test_worktree_worktree__validate(void)
 	git_worktree_free(wt);
 }
 
+void test_worktree_worktree__name(void)
+{
+	git_worktree *wt;
+
+	cl_git_pass(git_worktree_lookup(&wt, fixture.repo, "testrepo-worktree"));
+	cl_assert_equal_s(git_worktree_name(wt), "testrepo-worktree");
+}
```
Comment (tiennou) [inline tests/worktree/worktree.c]: Missing `git_worktree_free`.

File: tests/worktree/worktree.c [hunk 8]
```diff
@@ -355,6 +355,26 @@ void test_worktree_worktree__validate(void)
...
+{
+	git_worktree *wt;
+	git_buf expected_path = GIT_BUF_INIT;
+
+	cl_git_pass(git_buf_joinpath(&expected_path, clar_sandbox_path(), "testrepo-worktree"));
+	cl_git_pass(git_worktree_lookup(&wt, fixture.repo, "testrepo-worktree"));
+	cl_assert_equal_s(git_worktree_path(wt), expected_path.ptr);
+	
+	git_buf_free(&expected_path);
+}
```
Comment (tiennou) [inline tests/worktree/worktree.c]: Ditto.

**Comment by ethomson** [comment]
Don't have time to 👀 this just yet, but I wanted to make two quick comments from your prior PR:

> About /private/tmp, I think it's because the library rarely returns absolute paths (they're usually repo-root-relative), so your belief is correct 😉. I was just surprised it never came up.

This has definitely come up before, I'll have to look through some things for context.

> Closing this as I opened a new PR after consolidating some git history here: #4640

Generally it's easier for us to review one PR instead of two.  Feel free to iterate in the PR.

**Comment by mkeeler** [comment]
@ethomson Yeah, I wasn't so sure about how github would respond to a PR being updated with forced pushes with rewritten commits. It seems to work well enough though so I definitely should have done it for the other PR.

**Comment by pks-t** [comment]
Thanks again for this PR!

**Pull request merged**